src/components/ImageSlider.vue

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+export default {
+  data() {
+    return {
+      currentSlide: 0,
+      images: [
+        "/src/assets/image-product-1.jpg",
+        "/src/assets/image-product-2.jpg",
+        "/src/assets/image-product-3.jpg",
+        "/src/assets/image-product-4.jpg",
+      ],
+    };
+  },
+  methods: {
+    nextSlide() {
+      if (this.currentSlide < this.images.length - 1) {
+        this.currentSlide++;
+      } else {
+        this.currentSlide = 0;
+      }
+    },
+    previousSlide() {
+      if (this.currentSlide > 0) {
+        this.currentSlide--;
+      } else {
+        this.currentSlide = this.images.length - 1;
+      }
+    },
+  },
+};
+</script>
+<template>
+    <div class="relative w-full">
+      <div
+        class="w-full whitespace-nowrap transition-all duration-500"
+        :style="`transform: translateX(-${currentSlide * 100}%);`"
```

Comment:
Instead of manually computing the styles for the transform property, you can use a computed property to generate the styles dynamically:

```
computed: {
    slideStyles(): {
      return {
        transform: `translateX(-${this.currentSlide * 100}%)`,
      };
    },
  },
```